pulp_container/tests/functional/api/test_push_signatures.py

```diff
@@ -0,0 +1,119 @@
+import re
+import requests
+import subprocess
+import pytest
+
+from pulp_smash.pulp3.bindings import (
+    delete_orphans,
+    monitor_task,
+)
+from pulp_container.tests.functional.constants import REGISTRY_V2_REPO_PULP
+
+KEY = "https://raw.githubusercontent.com/pulp/pulp-fixtures/master/common/GPG-PRIVATE-KEY-pulp-qe"
+
+gpg = gnupg.GPG()
+
+
+@pytest.fixture(scope="session")
+def key_data():
+    """Return the ID and fingerprint of the Pulp QE GPG key."""
+    private_key = requests.get(KEY).text
+    with open("pulp-qe-private.key", "w") as f:
+        f.write(private_key)
+
+    subprocess.check_output(["sudo", "gpg", "--import", "pulp-qe-private.key"])
+
+    output = subprocess.check_output(
+        ["sudo", "gpg", "--keyid-format", "LONG", "-k", "Pulp QE"]
+    ).decode()
+    keyid = re.search(r"/([0-9A-F]+)", output).group(1)
+    fingerprint = re.search(r"([0-9A-F]+)\n", output).group(1)
+
+    subprocess.check_output(
+        ["sudo", "gpg", "--armor", "--output", "pulp-qe-public.key", "--export", "Pulp QE"]
+    )
+    with open("pulp-qe-public.key", "r") as f:
+        gpg.import_keys(f.read())
+
+    yield keyid, fingerprint
+
+    subprocess.check_output(
+        ["sudo", "gpg", "--batch", "--yes", "--delete-secret-keys", fingerprint]
+    )
+    subprocess.check_output(["sudo", "gpg", "--batch", "--yes", "--delete-keys", fingerprint])
+
+    gpg.delete_keys(fingerprint)
+
+    os.unlink("pulp-qe-public.key")
+    os.unlink("pulp-qe-private.key")
+
+
+@pytest.fixture
+def distribution(registry_client, local_registry, container_distribution_api, key_data):
+    """Return a distribution created after pushing a signed content to the Pulp Registry."""
+    registry_client.pull(f"{REGISTRY_V2_REPO_PULP}:manifest_a")
+
+    image_path = f"{REGISTRY_V2_REPO_PULP}:manifest_a"
+    local_image_path = "test-1:manifest_a"
+    identity, _ = key_data
+    local_registry.push(image_path, local_image_path, "--sign-by", identity)
+
+    distribution = container_distribution_api.list(name="test-1").results[0]
+
+    yield distribution
+
+    monitor_task(container_distribution_api.delete(distribution.pulp_href).task)
+    delete_orphans()
+
+
+def test_push_signed_image(
+    pulp_cfg,
+    local_registry,
+    container_push_repository_api,
+    container_manifest_api,
+    container_signature_api,
+    get_response,
+    key_data,
+    distribution,
+):
+    """Test whether an admin user can push signed images to the Pulp Registry."""
+    keyid, fingerprint = key_data
+
+    repository = container_push_repository_api.read(distribution.repository)
+    manifest = container_manifest_api.list(
+        repository_version=repository.latest_version_href
+    ).results[0]
+
+    signature = container_signature_api.list(
+        repository_version=repository.latest_version_href
+    ).results[0]
+
+    assert manifest.digest in signature.name
+    assert manifest.pulp_href == signature.signed_manifest
+    assert keyid == signature.key_id
```

Comment:
:+1:
Usually i write the "==" tests the other way around. On the left the thing that should be and on the right what it should be. Read like "assert that <something> is <value>".